### PR TITLE
boot: load bits of kernel command line from gadget snaps

### PR DESCRIPTION
**Pull request by bboozzoo**
Stacked on top of #10119, the relevant commit is: https://github.com/snapcore/snapd/commit/af2862f7e630911bb07b24272750a12998fb97d6

The branch adds support for loading bits of kernel command line from the gadget. Currently only cmdline.extra is supported.

### CODE REVIEW

File: boot/cmdline.go [hunk 1]
```diff
@@ -169,8 +169,25 @@ func composeCommandLine(model *asserts.Model, currentOrCandidate int, mode, syst
 		}
 		return "", err
 	}
-	// TODO:UC20: fetch extra args from gadget
 	extraArgs := ""
+	if gadgetDirOrSnapPath != "" {
+		sf, err := snapfile.Open(gadgetDirOrSnapPath)
```
Comment (pedronis) [inline boot/cmdline.go:174]: shouldn't we merge this Open into gadget.KernelCommandLineFromGadget ? do we have a case where we have the container already?

**Review by pedronis — APPROVED** [review]
thanks, one remark

File: boot/cmdline.go (hunk 1 above)
Comment (bboozzoo) [reply boot/cmdline.go:174]: Yes, that's a good point. I'll do it in a followup so that #10134  is unblocked first, if that's ok with you.

**Review by mvo5 — APPROVED** [review]
Thank you

**Comment by mvo5** [comment]
All failures look unreleated

**Pull request merged**